questionGenerator.py

```diff
@@ -0,0 +1,291 @@
+import pandas as pd
+from utils import dataLoader, dataSaver, printAndLog
+from config import OpenAIBot
+import json
+import sys
+
+class QuestionData:
+    """
+    Represents a class that handles question data generation.
+
+    Args:
+        config (Config): The configuration object.
+        videoData (VideoData): The video data object.
+        topicModeller (TopicModeller): The topic modeller object.
+        load (bool, optional): Whether to load existing question data. Defaults to True.
+    """
+
+    def __init__(self, config, videoData, topicModeller, load=True):
+        self.config = config
+
+        self.clusteredTopics = getClusteredTopics(videoData, topicModeller)
+        self.dominantTopics = getDominantTopic(self.clusteredTopics)
+        self.relevantText, self.questionText = getTextAndQuestion(
+            self.dominantTopics, videoData
+        )
+        self.responseData = None
+
+        if load:
+            self.loadQuestionData()
+        else:
+            self.OpenAIChatBot = OpenAIBot(self.config)
```

Comment:
Is there any reason why you are using OpenAIBot Instead of LangChainBot? As I understood you are using LangChain for topic extraction.

Reply:
This is because this is a much simpler task than the Topic Extraction part. Using LangChain in BERTopic lets BERTopic handle a lot of the complexities in how the topics are generated by leveraging LangChain, as it passes keywords, relevant documents, and such to get the human-interpretable topics.

In the case of Question generation, we only need to send one query per question we need to generate, and we get much finer control on the prompt directly because of it (LangChain is just a complicated way of building the prompt and storing information at its fundamental level as well, it just hides it from us through abstraction). So in this case, it made sense to stick to an OpenAIBot for our call rather than using LangChain.

Reply:
Please document that here and add link for further reading if there is any

Reply:
Addressed in the [new Commit](https://github.com/tl-its-umich-edu/annoto-gai/pull/16/commits/be278feff6e0c397bf70603df9d69149c0fcf026).

There isn't a link to document my exact implementation reasoning, so I put my explanation down there instead, but I did add a link to back up my reasoning for how LangChain is just a system atop OpenAI's prompting calls.